src/ralph/backends/storage/fs.py

```diff
@@ -0,0 +1,130 @@
+
+    def _get_filepath(self, name):
+        """Return path of the archive in the fsstorage, or throws an exception if not found"""
+        path = os.path.join(self._fspath, name)
+        if not os.path.isfile(path):
+            logger.error("%s does not exist", path)
+            raise FileNotFoundError()
+        return path
+
+    def _details(self, name):
+        """Get name archive details"""
+        path = self._get_filepath(name)
+        details = {}
+        details["filename"] = name
+        details["size"] = os.path.getsize(path)
+
+        return details
+
+    def list(self, details=False, new=False):
+        """List files in the storage backend"""
+        archives = os.listdir(self._fspath)
+        logger.debug("Found %d archives", len(archives))
+
+        if new:
+            archives = set(archives) - set(
+                entry.get("id")
+                for entry in filter(
+                    lambda e: e["backend"] == self.name and e["command"] == "fetch",
+                    self.history,
+                )
+            )
+            logger.debug("New archives: %d", len(archives))
+
+        for archive in archives:
+            yield self._details(archive) if details else archive
+
+    def url(self, name):
+        """Get `name` file absolute URL"""
+
+        path = self._get_filepath(name)
+        return path
+
+    def read(self, name, chunk_size=4096):
+        """Read `name` file and stream its content by chunks of a given size"""
+
+        logger.debug("Getting archive: %s", name)
+
+        path = self._get_filepath(name)
+        file = open(path, "rb")
+        chunk = file.read(chunk_size)
+        while chunk:
+            sys.stdout.buffer.write(chunk)
+            chunk = file.read(chunk_size)
+        file.close()
+
+        details = self._details(name)
+        # Archive is supposed to have been fully fetched, add a new entry to
+        # the history.
+        self.append_to_history(
+            {
+                "backend": self.name,
+                "command": "fetch",
+                "id": name,
+                "filename": details.get("filename"),
+                "size": details.get("size"),
+                "fetched_at": datetime.datetime.now(
+                    tz=datetime.timezone.utc
+                ).isoformat(),
+            }
+        )
+
+    def write(self, name, chunk_size=4096, overwrite=False):
+        """Write content to the `name` target"""
+
+        logger.debug("Creating archive: %s", name)
+
+        path = os.path.join(self._fspath, name)
+        if os.path.isfile(path) and not overwrite:
+            logger.error("%s already exists and overwrite not specified", name)
+            raise FileExistsError()
+
+        file = open(path, "w")
+        chunk = sys.stdin.read(chunk_size)
+        while chunk:
+            file.write(chunk)
+            chunk = sys.stdin.read(chunk_size)
+        file.close()
```

Comment:
Apply the same file opening pattern here (`with` statement and walrus operator).

Reply:
I didn't know about that walrus operator, it's fixed now